logger.go

```diff
@@ -187,47 +187,46 @@ func (log *Logger) With(fields ...Field) *Logger {
 // is enabled. It's a completely optional optimization; in high-performance
 // applications, Check can help avoid allocating a slice to hold fields.
 func (log *Logger) Check(lvl zapcore.Level, msg string) *zapcore.CheckedEntry {
-	return log.check(lvl, msg)
+	return log.check(lvl, msg, 0)
```

Comment:
I might be misreading, but should this be `1` instead? And for log methods below, should those be `3`? (There was previously a `const` offset of `2`, but we've since added another intermediate call.)

Can we add a test to validate the skips?